### DISCUSSION TITLE
Stringifying a function that references global variables (i.e. window, document, etc) results in a replacement of (void 0) when rendered SSR

### DISCUSSION THREAD

**Discussion by jonathan-rvezy** [nuxt/nuxt#27399] [Questions]
Here is the minimal example. https://stackblitz.com/edit/nuxt-starter-afu47n?file=nuxt.config.ts

```
const testFn = () => {
  console.log(window.console);
};

const fnString = String(testFn);
```

Will produce this when rendered SSR.

```
() => {
  console.log((void 0).console);
}
```

I'm guessing it's some kind of transpiling; however I really don't know where to go digging for this, so hoping someone can expedite answering this.  

If I put that into a test file and run

`npx tsx example.ts` it produces the desired output... so I don't believe it's something node related, but rather how nuxt is configured or one of its dependencies is configured/works.

**Comment by TheAlexLichter** [comment]
On the server side, `window` is replaced with undefined. This is intended as there is no `window` on the server. See eg. [this post](https://www.lichter.io/articles/universal-ssr-vue-component-guide/#window,-document,-and-friends---platform-specific-apis)

https://github.com/nuxt/nuxt/blob/d1ac12c98c86531c98f0e582e29f18774d498660/packages/vite/src/server.ts#L47

**Reply by jonathan-rvezy** [reply]
Right I understand that you can't use window server side, as well the window doesn't exist server side.

However this is running the javascript server side, its only intention is to generate a stringifed version for the client to use.

The use case here is that the javascript is being injected into the head inline rather than loading from a file.

If you have a suggestion for an alternative way to deal with that; I'm all ears.

Thanks

**Comment by jonathan-rvezy** [comment]
I ended up using globalThis which isn't replaced to undefined, and everything works as expected now.
